apollo-router/src/router_factory.rs

```diff
@@ -237,32 +230,8 @@ pub(crate) async fn create_subgraph_services(
                 subgraph::Request,
                 Response = subgraph::Response,
                 Error = BoxError,
-                Future = Either<
-                    Either<
-                        BoxFuture<'static, Result<subgraph::Response, BoxError>>,
-                        Either<
-                            BoxFuture<'static, Result<subgraph::Response, BoxError>>,
-                            timeout::future::ResponseFuture<
-                                Oneshot<
-                                    Either<
-                                        Retry<
-                                            RetryPolicy,
-                                            Either<
-                                                rate::service::RateLimit<SubgraphService>,
-                                                SubgraphService,
-                                            >,
-                                        >,
-                                        Either<
-                                            rate::service::RateLimit<SubgraphService>,
-                                            SubgraphService,
-                                        >,
-                                    >,
-                                    subgraph::Request,
-                                >,
-                            >,
-                        >,
-                    >,
-                    <SubgraphService as Service<subgraph::Request>>::Future,
+                Future = crate::plugins::traffic_shaping::TrafficShapingSubgraphFuture<
```

Comment:
omg thank you, i saw this yesterday and panicked at the idea that i might need to understand it